app/src/main/java/com/oatrice/internet_speed_testing/SpeedTestKotlin.kt

```diff
@@ -0,0 +1,115 @@
+package com.oatrice.internet_speed_testing
+
+import android.os.Bundle
+import android.support.v7.app.AppCompatActivity
+import com.example.internet_speed_testing.InternetSpeedBuilder
+import com.example.internet_speed_testing.ProgressionModel
+import kotlinx.android.synthetic.main.layout_speedtest_ui.*
+import java.math.BigDecimal
+import java.text.DecimalFormat
+
+class SpeedTestKotlin : AppCompatActivity() {
+
+    var internetSpeedBuilder:InternetSpeedBuilder?=null
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.layout_speedtest_ui)
+
+        internetSpeedBuilder = InternetSpeedBuilder(this)
+        setupView()
+    }
+
+    fun setupView(){
+        startButton.setOnClickListener {
+            buildTestSpeed()
+
+            if(startButton.text == "TEST AGAIN"){
+                downloadTextView.text = "0 mb/s"
+                uploadTextView.text = "0 mb/s"
+                totalTextView.text = "0 mb/s"
+
+                if(downloadTextView.text == "0 mb/s" && uploadTextView.text == "0 mb/s" && totalTextView.text == "0 mb/s"){
+                    internetSpeedBuilder = InternetSpeedBuilder(this)
+                    buildTestSpeed()
+                }
+
+            }
+        }
+    }
+
+    fun buildTestSpeed(){
+        startButton.isEnabled = false
+        internetSpeedBuilder?.setOnEventInternetSpeedListener(object: InternetSpeedBuilder.OnEventInternetSpeedListener{
+            override fun onProgressCompletion(boolean: Boolean, progressModel: ProgressionModel) {
+                if(progressModel.progressDownloadCompletion && progressModel.progressUploadCompletion){
+                    runOnUiThread {
+                        startButton.text = "TEST AGAIN"
```

Comment:
- Should show progressing state in button text or some things like that.
```suggestion
                        startButton.text = "Processing..."
```